numpy/core/src/multiarray/common.h

```diff
@@ -169,6 +179,11 @@ check_and_adjust_axis(int *axis, int ndim)
     }
     return 0;
 }
+static NPY_INLINE int
+check_and_adjust_axis(int *axis, int ndim)
+{
+    return check_and_adjust_axis_msg(axis, ndim, Py_None);
```

Comment:
Some confusion added here by my first comment being written before yours appeared (and not directed at you), I think :).